### DISCUSSION TITLE
Datahub, 2 strange errors

### DISCUSSION THREAD

**Discussion by BackyardRacing** [pimcore/pimcore#15465] [Q&A / Support] [answered]
Hi all

I'm testing with DataHub imports quite some time now. But can't get rid of following errors trying to import data from excel sheet.:

--------------------------

Issue1:
It only imports (and generates logs) for every second row. The rows in between are just ignored!!
Doesnt matter which rows, i can change their positions so the others get ignored and vice versa...

Tried with xslx or csv. Same results.

--------------------------

Issue2:
some elements cannot be importet. following error message can be found in the logs:

`Error processing element: "1234,XX,YY,SOME DATA,BLABLA" The "kernel" service is synthetic, it needs to be set at boot time before it can be used.`

Thanks for your help!!

**Comment by dvesh3** [comment]
Hey!! It would be nice if you can share the sample data to import and steps to reproduce. thanks!

**Reply by dvesh3** [reply]
and also the data importer version installed at your end

**Reply by BackyardRacing** [reply]
Hi, heres the csv:
https://www.dropbox.com/scl/fi/8s1vi4wznxmwrw90edo47/missing_fz.csv?rlkey=fnf0bw3hbhknhzockug8zr3xc&dl=0

this is the importer itself:
https://www.dropbox.com/scl/fi/l2x6baupyayeduqus6j3p/datahub_dataImporterDataObject_VEHimporter2_export.json?rlkey=qz9929z5isgcpjgf3ozjnr9b2&dl=0

PIMcore V11.0.0
with latest Datahub extension

**Reply by BackyardRacing** [reply]
Any news here?

**Reply by dvesh3** [reply]
@BackyardRacing please share class definitions as well.

**Reply by BackyardRacing** [reply]
https://www.dropbox.com/scl/fi/wqrl3t6lwzvu1d2zheptz/class_Vehicle_export.json?rlkey=o69rwkkr4jzme4tmgzkusgfd6&dl=0

**Reply by dvesh3** [reply, marked as answer]
So here are findings: 

1. In the import setting > Element Creation, you are using `static path` and System field "`KEY`" is mapped to `column [3] `which has duplicate values so objects are overwritten in this case and also you have folder rows in between, so you might need to use `"Find or Create Folder"` location strategy.
2. Issue 2 is not reproducible (please share you data importer version, might be fixed in the latest version).

**Reply by BackyardRacing** [reply]
Thanks for your info. In the meantime i've already tested alot more. There where no more duplicate values, a new coloumn for the ID and so on, solved ISSUE1. So all your findings are correct, but where already corrected, without having success in solving my problem ISSUE2...

But today i made a breaktrhough. I've tested affecting lines, changing field by field with some other data that is working in other lines in the csv. And there was one field which contains the link to an asset on the server. It came out that some of these assets (jpg images) where corrupt ("cannot read file header").
So it turned out that when i try to import a line which links to a corrupt asset, it just dont work! Without any error or warning against the real problem...

So maybe this helps, for generating more helpful errors or warnings in the logs, or when importing assets, or just somewhere... ;)

**Reply by BackyardRacing** [reply]
But now i try to delete those assets and have another problem right away...

Gives Error message, this are the details:

`{"success":false,"traceString":"in \/var\/www\/html\/vendor\/league\/flysystem\/src\/UnableToDeleteDirectory.php:27\n#0 \/var\/www\/html\/vendor\/league\/flysystem-local\/LocalFilesystemAdapter.php(160): League\\Flysystem\\UnableToDeleteDirectory::atLocation(\u00272\u0027, \u0027Unable to delet...\u0027)\n#1 \/var\/www\/html\/vendor\/league\/flysystem\/src\/Filesystem.php(90): League\\Flysystem\\Local\\LocalFilesystemAdapter-\u003EdeleteDirectory(\u00272\u0027)\n#2 \/var\/www\/html\/vendor\/pimcore\/pimcore\/models\/Asset.php(1648): League\\Flysystem\\Filesystem-\u003EdeleteDirectory(\u0027\/2\u0027)\n#3 \/var\/www\/html\/vendor\/pimcore\/pimcore\/models\/Asset.php(1723): Pimcore\\Model\\Asset-\u003EclearThumbnails(true)\n#4 \/var\/www\/html\/vendor\/pimcore\/pimcore\/models\/Asset.php(1034): Pimcore\\Model\\Asset-\u003EclearFolderThumbnails(Object(Pimcore\\Model\\Asset\\Folder))\n#5 \/var\/www\/html\/vendor\/pimcore\/admin-ui-classic-bundle\/src\/Controller\/Admin\/Asset\/AssetController.php(716): Pimcore\\Model\\Asset-\u003Edelete()\n#6 \/var\/www\/html\/vendor\/symfony\/http-kernel\/HttpKernel.php(166): Pimcore\\Bundle\\AdminBundle\\Controller\\Admin\\Asset\\AssetController-\u003EdeleteAction(Object(Symfony\\Component\\HttpFoundation\\Request))\n#7 \/var\/www\/html\/vendor\/symfony\/http-kernel\/HttpKernel.php(74): Symfony\\Component\\HttpKernel\\HttpKernel-\u003EhandleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)\n#8 \/var\/www\/html\/vendor\/symfony\/http-kernel\/Kernel.php(197): Symfony\\Component\\HttpKernel\\HttpKernel-\u003Ehandle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#9 \/var\/www\/html\/vendor\/symfony\/runtime\/Runner\/Symfony\/HttpKernelRunner.php(35): Symfony\\Component\\HttpKernel\\Kernel-\u003Ehandle(Object(Symfony\\Component\\HttpFoundation\\Request))\n#10 \/var\/www\/html\/vendor\/autoload_runtime.php(29): Symfony\\Component\\Runtime\\Runner\\Symfony\\HttpKernelRunner-\u003Erun()\n#11 \/var\/www\/html\/public\/index.php(19): require_once(\u0027\/var\/www\/html\/v...\u0027)\n#12 {main}","trace":[{"file":"\/var\/www\/html\/vendor\/league\/flysystem-local\/LocalFilesystemAdapter.php","line":160,"function":"atLocation","class":"League\\Flysystem\\UnableToDeleteDirectory","type":"::","args":["2","Unable to delete file at \/var\/www\/html\/public\/var\/tmp\/thumbnails\/2\/image-thumb__2__-folder-preview-hdpi.jpg"]},{"file":"\/var\/www\/html\/vendor\/league\/flysystem\/src\/Filesystem.php","line":90,"function":"deleteDirectory","class":"League\\Flysystem\\Local\\LocalFilesystemAdapter","type":"-\u003E","args":["2"]},{"file":"\/var\/www\/html\/vendor\/pimcore\/pimcore\/models\/Asset.php","line":1648,"function":"deleteDirectory","class":"League\\Flysystem\\Filesystem","type":"-\u003E","args":["\/2"]},{"file":"\/var\/www\/html\/vendor\/pimcore\/pimcore\/models\/Asset.php","line":1723,"function":"clearThumbnails","class":"Pimcore\\Model\\Asset","type":"-\u003E","args":[true]},{"file":"\/var\/www\/html\/vendor\/pimcore\/pimcore\/models\/Asset.php","line":1034,"function":"clearFolderThumbnails","class":"Pimcore\\Model\\Asset","type":"-\u003E","args":[{}]},{"file":"\/var\/www\/html\/vendor\/pimcore\/admin-ui-classic-bundle\/src\/Controller\/Admin\/Asset\/AssetController.php","line":716,"function":"delete","class":"Pimcore\\Model\\Asset","type":"-\u003E","args":[]},{"file":"\/var\/www\/html\/vendor\/symfony\/http-kernel\/HttpKernel.php","line":166,"function":"deleteAction","class":"Pimcore\\Bundle\\AdminBundle\\Controller\\Admin\\Asset\\AssetController","type":"-\u003E","args":[{"attributes":{},"request":{},"query":{},"server":{},"files":{},"cookies":{},"headers":{}}]},{"file":"\/var\/www\/html\/vendor\/symfony\/http-kernel\/HttpKernel.php","line":74,"function":"handleRaw","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"-\u003E","args":[{"attributes":{},"request":{},"query":{},"server":{},"files":{},"cookies":{},"headers":{}},1]},{"file":"\/var\/www\/html\/vendor\/symfony\/http-kernel\/Kernel.php","line":197,"function":"handle","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"-\u003E","args":[{"attributes":{},"request":{},"query":{},"server":{},"files":{},"cookies":{},"headers":{}},1,true]},{"file":"\/var\/www\/html\/vendor\/symfony\/runtime\/Runner\/Symfony\/HttpKernelRunner.php","line":35,"function":"handle","class":"Symfony\\Component\\HttpKernel\\Kernel","type":"-\u003E","args":[{"attributes":{},"request":{},"query":{},"server":{},"files":{},"cookies":{},"headers":{}}]},{"file":"\/var\/www\/html\/vendor\/autoload_runtime.php","line":29,"function":"run","class":"Symfony\\Component\\Runtime\\Runner\\Symfony\\HttpKernelRunner","type":"-\u003E","args":[]},{"file":"\/var\/www\/html\/public\/index.php","line":19,"args":["\/var\/www\/html\/vendor\/autoload_runtime.php"],"function":"require_once"}],"message":"Unable to delete directory located at: 2. Unable to delete file at \/var\/www\/html\/public\/var\/tmp\/thumbnails\/2\/image-thumb__2__-folder-preview-hdpi.jpg"}`

**Reply by dvesh3** [reply]
@BackyardRacing Can you please try to upload these assets on Admin UI and delete it? if it works there, then we need to fix it in data-hub extension.

**Reply by BackyardRacing** [reply]
They where already uploaded by drag-and-drop in the admin UI!
In the meantime our dev has found a problem with the right-management on the filesystem.
I think we will be able to solve it. 

But thank you for your time!
